File: Dagger2Demo/app/src/main/java/com/lyc/love/dagger2demo/MainActivity.java
package com.lyc.love.dagger2demo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

import com.google.gson.Gson;
import com.lyc.love.dagger2demo.bean.Goods;
import com.lyc.love.dagger2demo.bean.Order;
import com.lyc.love.dagger2demo.dagger2.component.DaggerMainActivityComponet;
import com.lyc.love.dagger2demo.dagger2.module.ObjectModule;
import com.lyc.love.dagger2demo.dagger2.scope.Name;

import javax.inject.Inject;


public class MainActivity extends AppCompatActivity {

    private static final String TAG="MainActivity";

    @Inject
    Goods mGoods;

    @Inject
    Goods mGoods2;

    @Inject
    Goods mGoods3;

    @Inject
    Order mOrder;

    @Name("value")
    @Inject
    Order mOrder2;

    @Inject
    Gson mGson;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerMainActivityComponet
                .builder()
                .appComponent(MyApplication.newInsatnce().getAppComponent())
                .objectModule(new ObjectModule())
                .build()
                .inject(this);

        Log.d(TAG,mGoods.toString());
        Log.d(TAG,mGoods2.toString());
        Log.d(TAG,mGoods3.toString());
        Goods goods=mOrder.getmGoods();
        Log.d(TAG,mOrder.toString());
        Log.d(TAG,goods.toString());
        Log.d(TAG,mGson.toString());
        Log.d(TAG,mOrder2.getName());


        findViewById(R.id.bt).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this,TwoActivity.class));
            }
        });
    }




}

File: MediaDemo/app/src/main/java/com/lyc/indonesia/mediademo/MainActivity.kt
package com.lyc.indonesia.mediademo

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.lyc.indonesia.mediademo.photo.CustomTakePictureActivity
import com.lyc.indonesia.mediademo.photo.CaputerActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
** 作者 梁永聪
** 时间 2019/6/8
 * 描述 多媒体模块总UI界面
**/
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        //系统拍照功能
        btn_camera.setOnClickListener {
            startActivity(Intent(this@MainActivity,CaputerActivity::class.java))
        }

        btn_custom_camera.setOnClickListener {
            startActivity(Intent(this@MainActivity,CustomTakePictureActivity::class.java))
        }
    }
}
